webui/react/src/types.ts

```diff
@@ -406,18 +398,32 @@ export interface Step extends WorkloadGroup, StartEndTimes {
   training: MetricsWorkload;
 }
 
-export interface Metrics {
-  numInputs?: number;
-  validationMetrics?: Record<string, number>;
+export interface Metrics extends Api.V1Metrics {
+  // these two fields are present in the protos
+  // as a struct and list of structs, respectively
+  // here, we are being a bit more precise
+  avgMetrics: Record<string, number>;
+  batchMetrics?: Array<Record<string, number>>;
 }
 
 export type Metadata = Record<RecordKey, string>;
 
-export interface CheckpointDetail extends Checkpoint {
-  batch: number;
+export interface CoreApiGenericCheckpoint {
```

Comment:
The purpose of the new name is to make the distinction between the types clear and to avoid the kind of entanglement from before:
- half the places checkpointDetail was being used before, it was not the right type for the data
- the new name makes clear the origin of the data